src/gt4py/next/program_processors/runners/dace/transformations/map_fusion_extended.py

```diff
@@ -459,6 +460,27 @@ def can_be_applied(
 
         # TODO(phimuell): Implement a check that the array is single use data?
 
+        # Check that the data read by the nested map are not accessed by any other outgoing edge
+        # of the access node.
+        access_node_out_edges = graph.out_edges(self.access_node)
+        access_node_out_edges_data = [edge.data.data for edge in access_node_out_edges]
+        second_map_in_edges = graph.in_edges(self.second_map_entry)
+        matching_data_subsets = {
+            second_map_edge.data.data: second_map_edge.data.subset
```

Comment:
OPTIONAL:
```suggestion
            second_map_edge.data.data: second_map_edge.data.src_subset
```
Using subset here is not wrong and I am sure that it would be correct all the times, but using `src_subset` is in view a bit cleaner as it is immediately clear what side you want to access.